### PR TITLE
added the possibility to issue transactions in batch

### PR DESCRIPTION
**Pull request by Joachim-Lebrun**

### CODE REVIEW

File: contracts/token/TransferManager.sol [hunk 1]
```diff
@@ -157,6 +157,12 @@ contract TransferManager is Pausable {
         revert("Transfer not possible");
     }
 
+    function batchTransfer(address[] _to, uint256[] _value) external {
```
Comment (Nakasar) [inline contracts/token/TransferManager.sol]: batchTransfer doesn't make sense I think (an individual holder would never call this method).

Comment (Nakasar) [reply contracts/token/TransferManager.sol]: Nevermind, it can, why not.

Comment (Joachim-Lebrun) [reply contracts/token/TransferManager.sol]: yes, it's a request that we had from some issuers, they want to mint the total supply of tokens on their wallet and then proceed to the distribution by using the `transfer` function

**Comment by Joachim-Lebrun** [comment]
we need to wait for @dilipror to write the tests before merging this PR

**Comment by dilipror** [comment]
> we need to wait for @dilipror to write the tests before merging this PR

Done test cases

File: contracts/registry/IdentityRegistry.sol [hunk 2]
```diff
@@ -49,6 +49,12 @@ contract IdentityRegistry is IIdentityRegistry, AgentRole {
         emit IdentityRegistered(_user, _identity);
     }
 
+    function batchRegisterIdentity(address[] calldata _user, Identity[] calldata _identity, uint16[] calldata _country) external {
```
Comment (Nakasar) [inline contracts/registry/IdentityRegistry.sol]: Maybe use plural form for arguments.

```suggestion
    function batchRegisterIdentity(address[] calldata _users, Identity[] calldata _identities, uint16[] calldata _countries) external {
```

File: contracts/registry/IdentityRegistry.sol [hunk 3]
```diff
@@ -49,6 +49,12 @@ contract IdentityRegistry is IIdentityRegistry, AgentRole {
...
+        for (uint256 i = 0; i < _user.length; i++) {
```
Comment (Nakasar) [inline contracts/registry/IdentityRegistry.sol]:
```suggestion
        for (uint256 i = 0; i < _users.length; i++) {
```

File: contracts/registry/IdentityRegistry.sol [hunk 4]
```diff
@@ -49,6 +49,12 @@ contract IdentityRegistry is IIdentityRegistry, AgentRole {
...
+            registerIdentity(_user[i], _identity[i], _country[i]);
```
Comment (Nakasar) [inline contracts/registry/IdentityRegistry.sol]:
```suggestion
            registerIdentity(_users[i], _identities[i], _countries[i]);
```

File: contracts/token/TransferManager.sol [hunk 5]
```diff
@@ -343,27 +355,39 @@ contract TransferManager is Pausable {
...
     /**
      *  Freezes token amount specified for given address.
      *  @param addr The address for which to update frozen tokens
      *  @param amount Amount of Tokens to be frozen
      */
     function freezePartialTokens(address addr, uint256 amount)
-        onlyAgent
-        external
+        public
+		onlyAgent
```
Comment (Nakasar) [inline contracts/token/TransferManager.sol]: indentation issue here.

File: test/registry.test.js [hunk 6]
```diff
@@ -181,6 +181,15 @@ contract("IdentityRegistry", accounts => {
     const result = await identityRegistry.contains(accounts[3]);
     result.should.equal(false);
   });
+
+  it("Should register identities in batch", async () => {
+	claimHolder3 = await ClaimHolder.new({ from: accounts[3] });
```
Comment (Nakasar) [inline test/registry.test.js]: Check you identation.

File: test/registry.test.js [hunk 7]
```diff
@@ -181,6 +181,15 @@ contract("IdentityRegistry", accounts => {
...
+    await identityRegistry.batchRegisterIdentity(
+      [accounts[2], accounts[3]],
+      [claimHolder2.address, claimHolder3.address],
+	  [91, 101]
+    );
+  });
```
Comment (Nakasar) [inline test/registry.test.js]: You should verify that Identities were added by querying them after execution the batch operation, here you only verify that the transaction was a success.

File: truffle.js [hunk 8]
```diff
@@ -53,7 +53,13 @@ module.exports = {
   },
   compilers: {
     solc: {
-      version: '0.5.10'
+	  version: '0.5.10',
```
Comment (Nakasar) [inline truffle.js]: Wrong indentation here?

**Comment by Joachim-Lebrun** [comment]
> Arguments should be put in plural form, and caution with code indentation

done

**Comment by Nakasar** [comment]
I think parameters in other batch methods should also be set to plural form

**Comment by Joachim-Lebrun** [comment]
> Arguments should be put in plural form, and caution with code indentation



> I think parameters in other batch methods should also be set to plural form

done

File: test/registry.test.js (hunk 7 above)
Comment (Joachim-Lebrun) [reply test/registry.test.js]: done, for this one and all others

**Pull request merged**